ocaml/typing/types.ml

```diff
@@ -316,8 +316,15 @@ and constructor_declaration =
     cd_uid: Uid.t;
   }
 
+and constructor_argument =
+  {
+    ca_global: Mode.Global_flag.t;
+    ca_type: type_expr;
+    ca_loc: Location.t;
```

Comment:
Is `ca_loc` only meaningful if `ca_global` is `Global`? I say this because it looks like `ca_loc` is ignored in `tast_iterator`, which is not clearly OK unless the property I state is true. You might want to comment that `ca_loc` is only used for untypeast, and is only set to a non-none value if `ca_global` is `Global` (or whatever property is actually true).

Reply:
Not sure I understand your point about `tast_iterator`, since that operates on `Typedtree`, not `Types`. Maybe you meant to comment this on the parallel record there? But that field isn't ignored in `tast_iterator`. Honestly the only reason I added a loc field to both these records is because it seemed to me that every other record in `Parsetree` that had a loc there also has a loc in `Typedtree` and `Types`.

Reply:
Yes, I meant to leave this comment on `Typedtree`.

The thing I'm responding to is the fact that it appears that `ca_loc` is sometimes set to a dummy value, `Location.none`. It would be nice if that class of cases were captured in a comment. I believe the class of cases is just "`ca_global` is Unrestricted".

Reply:
It also might be more explicit to make this a `Location.t option` that's populated only when the other field is `Global`. (That way e.g. `tast_iterator` isn't iterating over a none location sometimes. I don't think this is actually problematic, but it's slightly odd.)

Reply:
Ah, OK, I see another layer of my confusion. I didn't realize there were two different locations (the one on the global flag and the one on the constructor arg itself.) I'm going to resolve this. If I have further thoughts I'll open another thread.